CommandDotNet/Models/AppSettings.cs

```diff
@@ -31,5 +31,8 @@ public AppSettings()
         internal IHelpProvider CustomHelpProvider { get; set; }
 
         internal ICommandInvoker CommandInvoker { get; set; }
+
+        // lives here so the same instance is used everywhere.  other choices were messy.
+        internal CachedArgumentModelResolver ArgumentModelResolver { get; set; }
```

Comment:
This would have been a much larger PR if I didn't put this here.  I played around a bit trying to find how to get the single instance passed into the correct places but there didn't seem to be a cleaner way.